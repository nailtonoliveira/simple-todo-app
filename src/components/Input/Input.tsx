export interface InputProps
  extends React.InputHTMLAttributes<HTMLInputElement> {}

export default function Input({ className, ...props }: InputProps) {
  const classes = [
    "block",
    "rounded-md",
    "border-0",
    "py-1.5",
    "px-4",
    "text-sm",
    "leading-6",
    "text-zinc-50",
    "placeholder:text-zinc-400",
    "ring-1",
    "ring-inset",
    "ring-zinc-700",
    "focus:ring-2",
    "focus:ring-inset",
    "focus:ring-orange-600",
    "focus:outline-none",
    "bg-zinc-800",
    "shadow-sm",
    "transition",
    "duration-500",
    "ease-in-out",
  ];

  return <input className={[...classes, className].join(" ")} {...props} />;
}
