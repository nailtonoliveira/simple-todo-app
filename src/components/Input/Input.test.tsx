import { expect, test } from "vitest";
import { render } from "@testing-library/react";

import Input from "./Input";

test("Input component", () => {
  const result = render(<Input />);

  expect(result).toMatchSnapshot();
});
