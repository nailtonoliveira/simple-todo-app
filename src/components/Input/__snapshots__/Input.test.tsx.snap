// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="block rounded-md border-0 py-1.5 px-4 text-sm leading-6 text-zinc-900 placeholder:text-zinc-400 ring-1 ring-inset ring-zinc-700 focus:ring-2 focus:ring-inset focus:ring-orange-600 focus:outline-none bg-zinc-800 shadow-sm "
      />
    </div>
  </body>,
  "container": <div>
    <input
      class="block rounded-md border-0 py-1.5 px-4 text-sm leading-6 text-zinc-900 placeholder:text-zinc-400 ring-1 ring-inset ring-zinc-700 focus:ring-2 focus:ring-inset focus:ring-orange-600 focus:outline-none bg-zinc-800 shadow-sm "
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
